tests/unit/frameworks/vernon/multi_cycle_lr_test.py

```diff
@@ -0,0 +1,166 @@
+        (
+            0,
+            dict(
+                max_lr=1.5,
+                pct_start=0.2,
+                anneal_strategy="linear",
+                base_momentum=0.6,
+                max_momentum=0.75,
+                cycle_momentum=True,
+                div_factor=6.0,
+                final_div_factor=1000.0,
+            ),
+        ),
+        (
+            1,
+            dict(
+                max_lr=1.0,
+                pct_start=0.2,
+                anneal_strategy="cos",
+                base_momentum=0.55,
+                max_momentum=0.7,
+                cycle_momentum=True,
+                div_factor=3.0,
+                final_div_factor=1000.0,
+            ),
+        ),
+        (
+            5,
+            dict(
+                max_lr=0.05,
+                pct_start=0.2,
+                anneal_strategy="linear",
+                base_momentum=0.6,
+                max_momentum=0.75,
+                cycle_momentum=False,
+                div_factor=25.0,
+                final_div_factor=100.0,
+            ),
+        ),
+    ),
+)
+
+
+class MultiCycleLRTest(unittest.TestCase):
+    def test_supervised_experiment_with_cycle_lr(self):
+
+        # Setup experiment and initialize model.
+        exp = supervised_experiment["experiment_class"]()
+        exp.setup_experiment(supervised_experiment)
+
+        num_images = len(exp.train_loader.dataset)
+        num_batches = -(-num_images // exp.train_loader.batch_size)
+
+        # Loop through some pseudo epochs.
+        cycle = 0
+        for i in range(exp.epochs):
+            lr = []
+
+            exp.pre_epoch()
+            # Loop through batches and get corresponding lr.
+            for batch in range(num_batches):
+                exp.pre_batch(exp.model, batch)
+                exp.optimizer.step()
+                lr.append(exp.get_lr()[0])
+                exp.post_batch(
+                    exp.model, 0, None, batch, exp.train_loader.batch_size, ""
+                )
+            exp.post_epoch()
+            exp.current_epoch += 1
+
+            # Get the current max lr according to the multiCycleLR config
+            if (
+                cycle < len(supervised_experiment["multi_cycle_lr_args"])
+                and supervised_experiment["multi_cycle_lr_args"][cycle][0] == i
+            ):
+                current_lr_conf = supervised_experiment["multi_cycle_lr_args"][
+                    cycle
+                ][1]
+                cycle += 1
+                # If new cycle is started, check that initial lr is
+                # max_lr/div_factor
+                initial_lr = (
+                    current_lr_conf["max_lr"] / current_lr_conf["div_factor"]
+                )
+                self.assertAlmostEqual(lr[0], initial_lr)
+
+            # Check that lr of this epoch stays below max lr according to
+            # multiCycleLR schedule.
+            assert np.max(lr) <= current_lr_conf["max_lr"]
```

Comment:
Unless I'm missing something, we shouldn't use assert's here - should use some form of self.assert...()

Reply:
Good point. 
Using [self.assert...()](https://docs.python.org/3/library/unittest.html#assert-methods) instead of plain `assert` allow the runner to accumulate the results

Reply:
Alright, I updated this: https://github.com/numenta/nupic.research/pull/620/commits/5e1d69f99dd42247df9f4b96db254d7c30f7ba22